### DISCUSSION TITLE
PVcell - saturation current

### DISCUSSION THREAD

**Discussion by lp74** [SunPower/PVMismatch#150] [General]
Hello,

I've got a question about the code of the saturation currents Isat1 and Isat2 of the pvcell.py.

PVMismatch uses the following formulation:

<img src="https://render.githubusercontent.com/render/math?math=\frac{I_{sat, i}}{I_{sat, i}^{*}} = \left(\frac{T_{c}}{T_c^*}\right)^3 \exp \left[ \frac{q \cdot E_g}{a_i \cdot k} \cdot \left( \frac{1}{T_c^*} -\frac{1}{T_c} \right)\right]">

That is reported in many papers, and where a_i is 1, 2 for diode 1 and 2 respectively.

What is really strange with this formulation is that the unit of the argument of the exp() is the charge (Coulomb).

Physically speaking, this is inconsistent.

I've developed the steps to compute the formula from the expression given in the site pv education and I've found what follow:

<img src="https://render.githubusercontent.com/render/math?math=\frac{I_{sat, i}}{I_{sat, i}^{*}} = \left(\frac{T_{c}}{T_c^*}\right)^3 \exp \left[ \frac{E_g}{a_i \cdot k} \cdot \left( \frac{1}{T_c^*} -\frac{1}{T_c} \right)\right]">

So my question is: do I have spotted an error or there is an explanation?

**Comment by mikofski** [comment]
Hi, Thanks for starting this discussion!

**TL;DR**: No, I do not think you've spotted an error because [the conversion factor from eV to Joules is q (or 1.6e-19[J/eV])](https://en.wikipedia.org/wiki/Electronvolt#Definition):
>An electronvolt ... has a value of one volt, 1 J/C, multiplied by the electron's elementary charge [q], 1.602176634×10−19 C. Therefore, one electronvolt is equal to 1.602176634×10−19 J.

I'm rerendering the equations here with a white background b/c I use the dark theme and can't see them, please check if they're what you intended. I copy/pasted them directly to [codecogs](https://www.codecogs.com/eqnedit.php)

### existing equation:
<a href="https://www.codecogs.com/eqnedit.php?latex=\bg_white&space;\frac{I_{sat,&space;i}}{I_{sat,&space;i}^{*}}&space;=&space;\left(\frac{T_{c}}{T_c^*}\right)^3&space;\exp&space;\left[&space;\frac{q&space;\cdot&space;E_g}{a_i&space;\cdot&space;k}&space;\cdot&space;\left(&space;\frac{1}{T_c^*}&space;-\frac{1}{T_c}&space;\right)\right]" target="_blank"><img src="https://latex.codecogs.com/png.latex?\bg_white&space;\frac{I_{sat,&space;i}}{I_{sat,&space;i}^{*}}&space;=&space;\left(\frac{T_{c}}{T_c^*}\right)^3&space;\exp&space;\left[&space;\frac{q&space;\cdot&space;E_g}{a_i&space;\cdot&space;k}&space;\cdot&space;\left(&space;\frac{1}{T_c^*}&space;-\frac{1}{T_c}&space;\right)\right]" title="\frac{I_{sat, i}}{I_{sat, i}^{*}} = \left(\frac{T_{c}}{T_c^*}\right)^3 \exp \left[ \frac{q \cdot E_g}{a_i \cdot k} \cdot \left( \frac{1}{T_c^*} -\frac{1}{T_c} \right)\right]" /></a>

### proposed equation
<a href="https://www.codecogs.com/eqnedit.php?latex=\bg_white&space;\frac{I_{sat,&space;i}}{I_{sat,&space;i}^{*}}&space;=&space;\left(\frac{T_{c}}{T_c^*}\right)^3&space;\exp&space;\left[&space;\frac{E_g}{a_i&space;\cdot&space;k}&space;\cdot&space;\left(&space;\frac{1}{T_c^*}&space;-\frac{1}{T_c}&space;\right)\right]" target="_blank"><img src="https://latex.codecogs.com/png.latex?\bg_white&space;\frac{I_{sat,&space;i}}{I_{sat,&space;i}^{*}}&space;=&space;\left(\frac{T_{c}}{T_c^*}\right)^3&space;\exp&space;\left[&space;\frac{E_g}{a_i&space;\cdot&space;k}&space;\cdot&space;\left(&space;\frac{1}{T_c^*}&space;-\frac{1}{T_c}&space;\right)\right]" title="\frac{I_{sat, i}}{I_{sat, i}^{*}} = \left(\frac{T_{c}}{T_c^*}\right)^3 \exp \left[ \frac{E_g}{a_i \cdot k} \cdot \left( \frac{1}{T_c^*} -\frac{1}{T_c} \right)\right]" /></a>

AFAICT the equation we are trying to represent is `qV/(nkT)` which has units `[A*s]*[V] / ( [J/K] * [K] )` where a Coulomb is represented as an Ampere-second. This expression simplifies to unity b/c `[A*V] = [W]` and `[W*s] = [J]` in the numerator and the denominator is also Joules b/c Kelvin cancels out.

The proposed equation would assume that the bandgap energy were already given in Joules, which is a fair assumption. However in PVMismatch, `Eg` is given in electron-volts, not Joules, therefore `q` is necessary to convert it to Joules.

Perhaps there's something that could be added to the documentation to make sure users know to use electron-volts and not Joules. I think that would be a really good improvement. Thanks for bringing this up!

**Reply by lp74** [reply]
thank you for the explanation
